modules/cudf/src/column.ts

```diff
@@ -700,6 +701,98 @@ export interface Column<T extends DataType = any> {
    */
   isNotNaN(memoryResource?: MemoryResource): Column<Bool8>;
 
+  /**
+   * Creates a column of `BOOL8` elements indicating strings in which all characters are valid for
+   * conversion to floats.
+   *
+   * @param memoryResource The optional MemoryResource used to allocate the result Column's device
+   *   memory.
+   * @returns A non-nullable column of `BOOL8` elements with `true` representing convertible
+   *   values
+   */
+  isFloat(memoryResource?: MemoryResource): Column<Bool8>;
+
+  /**
+   * Returns a new strings column converting the float values from the provided column into strings.
+   *
+   * Any null entries will result in corresponding null entries in the output column.
+   *
+   * For each float, a string is created in base-10 decimal. Negative numbers will include a '-'
+   * prefix. Numbers producing more than 10 significant digits will produce a string that includes
+   * scientific notation (e.g. "-1.78e+15").
+   *
+   * @param memoryResource The optional MemoryResource used to allocate the result Column's device
+   *   memory.
+   *
+   *  @returns A string Column with floats as strings.
+   */
+  fromFloats(memoryResource?: MemoryResource): Column<Utf8String>;
+
+  /**
+   * Returns a new floating point numeric column parsing float values from the provided
+   * strings column.
+   *
+   * Any null entries will result in corresponding null entries in the output column.
+   *
+   * Only characters [0-9] plus a prefix '-' and '+' and decimal '.' are recognized. Additionally,
+   * scientific notation is also supported (e.g. "-1.78e+5").
+   *
+   * @param dataType Type of floating numeric column to return.
+   * @param memoryResource  The optional MemoryResource used to allocate the result Column's device
+   *   memory.
+   *
+   *  @returns A Column of a the specified float type with the results of the conversion.
+   */
+  toFloats<R extends DataType>(dataType: R, memoryResource?: MemoryResource): Column<R>;
```

Comment:
only suggestion w/ this and `fromFloats` is maybe update the name to indicate it's for strings? This is in the `cudf::strings` namespace so the behavior is clear, but here it seems like we could use it for any dtype?

```suggestion
  stringsToFloats<R extends DataType>(dataType: R, memoryResource?: MemoryResource): Column<R>;
```

Reply:
ah yeah, I spoke with Bryan about my confusion that this was for strings, I think this is probably good solution.

Reply:
Seems reasonable, I was just hewing the the existing libcudf names as a default